### DISCUSSION TITLE
Postgres on VMs while etcd in kubernetes

### DISCUSSION THREAD

**Discussion by ahussainrc** [autobase-tech/autobase#661] [Q&A]
**My Setup:**
- Postgres servers are Virtual Machines while etcd is deployed in kubernetes.  
- 5 Pods for etcd are running with PVs.
- There is only one service with load balancer IPs to access etcd.
- Postgres servers can access IP of k8s service because all VMs (kubernetes nodes as well as  postgres nodes) are on-prem.

**Question:**
Below section of main.yml file expects host and port of all etcd servers but in my case etcd servers are kubernetes pods.
```
# if dcs_type: "etcd" and dcs_exists: true
patroni_etcd_hosts: []  # list of servers of an existing etcd cluster
#  - { host: "10.128.64.140", port: "2379" }
#  - { host: "10.128.64.142", port: "2379" }
#  - { host: "10.128.64.143", port: "2379" }
```
**Can I just add k8s service IP  in etcd host list ? 
For Patroni, there will be only one host for etcd (not sure if it works).**

_**Additional Notes:**_
_1. Reason to setup etcd is to avoid creating and maintaining separate VMs for etcd as k8s cluster is already there, and I can have large no of replicas for etcd._
_2. Postgres servers are NOT in kuberenetes on purpose._

**Comment by vitabaks** [comment]
@ahussainrc postgresql_cluster does not support integration with Kubernetes

most likely you are interested in the [kubernetes](https://patroni.readthedocs.io/en/latest/yaml_configuration.html#kubernetes) option and not [etcd3](https://github.com/vitabaks/postgresql_cluster/blob/master/roles/patroni/templates/patroni.yml.j2#L38), which is used for the `patroni_etcd_hosts` variable.

You can try to specify the options you need directly in the configuration template file - https://github.com/vitabaks/postgresql_cluster/blob/master/roles/patroni/templates/patroni.yml.j2

**Reply by ahussainrc** [reply]
@vitabaks 
Thanks for your reply.

[kubernetes](https://patroni.readthedocs.io/en/latest/yaml_configuration.html#kubernetes) section is for the scenario when  Patroni and postgres both are deployed in Kubernetes OR atleast Patroni is deployed in kubernetes. 

In my case, I have deployed patroni on postgres nodes, however I just want to deploy etcd in kubernetes so that Patroni can use it.

**Reply by vitabaks** [reply]
I have not had the practice of such an installation, so I cannot help you with this issue at the moment.
